packages/teams-js/src/internal/utils.ts

```diff
@@ -266,3 +266,33 @@ export function callCallbackWithErrorOrResultOrNullFromPromiseAndReturnPromise<T
   });
   return p;
 }
+
+/**
+ * A helper function to add a timeout to an asynchronous operation.
+ *
+ * @param action Action to wrap the timeout around
+ * @param timeoutInMs Timeout period
+ * @param timeoutError Error to reject the promise with if timeout elapses before the action completed
+ * @returns A promise which resolves to the result of provided action or rejects with a provided timeout error
+ * if the initial action didn't complete within provided timeout.
+ *
+ * @internal
+ */
+export function runWithTimeout<TResult, TError>(
+  action: () => Promise<TResult>,
+  timeoutInMs: number,
+  timeoutError: TError,
+): Promise<TResult> {
+  return new Promise((resolve, reject) => {
+    const timeoutHandle = window.setTimeout(reject, timeoutInMs, timeoutError);
```

Comment:
I don't know a ton about promises or timeout functions so this is a learning opportunity for me- From reading the docs on [setTimeout()](https://developer.mozilla.org/en-US/docs/Web/API/setTimeout), the 'timeoutError' argument seems to be passed into 'action', but the example we see here doesn't seem to use the argument and I can't find elsewhere that timeoutError gets used. Did I understand what's going on correctly? If so, did you want to use timeoutError somewhere?

Reply:
Everything after the second parameter will be pushed as arguments to the first parameter when the timer fires.

```
window.setTimeout(reject, timeoutInMs, timeoutError);
``` 

is basically equivalent to

```
window.setTimeout(() => reject(timeoutError), timeoutInMs)
```

but it's saving a closure allocation and level of indirection with the new lambda.

Reply:
Awesome, thank you :)